### PR TITLE
chore: fix yarn webpack 5

### PR DESCRIPTION
**Pull request by AlexVarchuk**
## What/Why/How?
fixes: https://github.com/Redocly/redoc/issues/2061
fixes: https://github.com/Redocly/redoc/issues/1820
This PR related to https://github.com/Redocly/redoc/issues/1820#issuecomment-1166599837. 
update webpack for fix bug in `loader-runner` (throw new Error("callback(): The callback was already called.");)
update openapi-core to v1.0.0-beta.104

step to reproduce:

- versions:
  nodejs: v18.4.0
  yarn 1.22.19
 
- steps:
  yarn install
  yarn run build:demo
 
## Check yourself

- [x] Code is linted
- [x] Tested
- [ ] All new/updated code is covered with tests

### CODE REVIEW

**Comment by deancn** [comment]
Arch: Apple M1
NodeJS: 18.4.0

use the branch,  and still cannot run build:demo.

**Comment by AlexVarchuk** [comment]
@deancn Hi. We know that. It is still draft PR. We wait the next openapi-core release.

**Comment by deancn** [comment]
@AlexVarchuk   Thank you so much.

**Comment by deancn** [comment]
> Hi @deancn. Can you verify it works?

M1 v18.5.0 :  success.

But Linux v18.6.0 still fail,  error msg:

```text
assets by status 1.08 MiB [cached] 4 assets
orphan modules 1020 KiB [orphan] 217 modules
runtime modules 1.29 KiB 7 modules
modules by path ./node_modules/ 1.38 MiB
  javascript modules 1.38 MiB 389 modules
  json modules 5.22 KiB
    modules by path ./node_modules/@redocly/ajv/dist/ 3.07 KiB
      ./node_modules/@redocly/ajv/dist/refs/json-schema-draft-07.json 2.72 KiB [built] [code generated]
      ./node_modules/@redocly/ajv/dist/refs/data.json 360 bytes [built] [code generated]
    ./node_modules/swagger2openapi/package.json 1.05 KiB [built] [code generated]
    ./node_modules/@redocly/openapi-core/package.json 1.1 KiB [built] [code generated]
modules by path ./src/ 31.8 KiB
  ./src/polyfills.ts 61 bytes [built] [code generated]
  ./node_modules/workerize-loader/dist/index.js?inline&fallback=false!./src/services/SearchWorker.worker.ts 31.7 KiB [not cacheable] [built] [code generated]
+ 11 modules
  Entrypoint HtmlWebpackPlugin_0-0 =
  runtime modules 440 bytes 3 modules
  cacheable modules 1.84 KiB
    data:text/javascript,__webpack_public.. 77 bytes [built] [code generated]
    ./node_modules/html-webpack-plugin/lib/loader.js!./demo/index.html 1.76 KiB [built] [code generated]

  Child HtmlWebpackCompiler compiled successfully

  assets by status 30.8 KiB [cached] 1 asset
  runtime modules 663 bytes 3 modules
  cacheable modules 101 KiB
    ./node_modules/workerize-loader/dist/rpc-worker-loader.js!./node_modules/esbuild-loader/dist/index.js??ruleSet[1].rules[1]!./src/services/SearchWorker.worker.ts 3.49 KiB [built] [code generated]
    ./node_modules/lunr/lunr.js 97.5 KiB [built] [code generated]

  Child worker compiled successfully

ERROR in cli/index.ts:343:47
TS2345: Argument of type 'CElement<{ store: any; }, Component<{ store: any; }, any, any>>' is not assignable to parameter of type 'ReactNode'.
  Property 'children' is missing in type 'ComponentElement<{ store: any; }, Component<{ store: any; }, any, any>>' but required in type 'ReactPortal'.
    341 |     const store = await createStore(spec, specUrl, redocOptions);
    342 |     const sheet = new ServerStyleSheet();
  > 343 |     html = renderToString(sheet.collectStyles(React.createElement(Redoc, { store })));
        |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    344 |     css = sheet.getStyleTags();
    345 |     state = await store.toJS();
    346 |

ERROR in src/common-elements/shelfs.tsx:37:33
TS2769: No overload matches this call.
  Overload 1 of 2, '(component: AnyStyledComponent): ThemedStyledFunction<any, ResolvedThemeInterface, any, any>', gave the following error.
    Argument of type 'typeof IntShelfIcon' is not assignable to parameter of type 'AnyStyledComponent'.
      Type 'typeof IntShelfIcon' is not assignable to type 'StyledComponent<any, any, any, never>'.
        Type 'typeof IntShelfIcon' is not assignable to type 'string'.
  Overload 2 of 2, '(component: "symbol" | "object" | "template" | "title" | "input" | "a" | "abbr" | "address" | "area" | "article" | "aside" | "audio" | "b" | "base" | "bdi" | "bdo" | "big" | "blockquote" | "body" | ... 157 more ... | FunctionComponent<...>): ThemedStyledFunction<...>', gave the following error.
    Argument of type 'typeof IntShelfIcon' is not assignable to parameter of type '"symbol" | "object" | "template" | "title" | "input" | "a" | "abbr" | "address" | "area" | "article" | "aside" | "audio" | "b" | "base" | "bdi" | "bdo" | "big" | "blockquote" | "body" | ... 157 more ... | FunctionComponent<...>'.
      Type 'typeof IntShelfIcon' is not assignable to type 'ComponentClass<any, any>'.
        Types of property 'contextType' are incompatible.
          Type 'React.Context<any> | undefined' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Context<any> | undefined'.
            Type 'React.Context<any>' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Context<any>'.
              The types of 'Provider.propTypes' are incompatible between these types.
                Type 'React.WeakValidationMap<React.ProviderProps<any>> | undefined' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").WeakValidationMap<import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").ProviderProps<any>> | undefined'.
                  Type 'React.WeakValidationMap<React.ProviderProps<any>>' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").WeakValidationMap<import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").ProviderProps<any>>'.
                    Types of property 'children' are incompatible.
                      Type 'React.Validator<React.ReactNode> | undefined' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Validator<import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").ReactNode> | undefined'.
                        Type 'React.Validator<React.ReactNode>' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Validator<import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").ReactNode>'.
                          Type 'React.ReactNode' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").ReactNode'.
                            Type '{}' is not assignable to type 'ReactNode'.
                              Type '{}' is missing the following properties from type 'ReactPortal': key, children, type, props
    35 | }
    36 |
  > 37 | export const ShelfIcon = styled(IntShelfIcon)`
       |                                 ^^^^^^^^^^^^
    38 |   height: ${props => props.size || '18px'};
    39 |   width: ${props => props.size || '18px'};
    40 |   min-width: ${props => props.size || '18px'};

ERROR in src/components/Callbacks/styled.elements.ts:5:43
TS2769: No overload matches this call.
  Overload 1 of 2, '(component: AnyStyledComponent): ThemedStyledFunction<any, ResolvedThemeInterface, any, any>', gave the following error.
    Argument of type 'typeof CallbackTitle' is not assignable to parameter of type 'AnyStyledComponent'.
      Type 'typeof CallbackTitle' is not assignable to type 'StyledComponent<any, any, any, never>'.
        Type 'typeof CallbackTitle' is not assignable to type 'string'.
  Overload 2 of 2, '(component: "symbol" | "object" | "template" | "title" | "input" | "a" | "abbr" | "address" | "area" | "article" | "aside" | "audio" | "b" | "base" | "bdi" | "bdo" | "big" | "blockquote" | "body" | ... 157 more ... | FunctionComponent<...>): ThemedStyledFunction<...>', gave the following error.
    Argument of type 'typeof CallbackTitle' is not assignable to parameter of type '"symbol" | "object" | "template" | "title" | "input" | "a" | "abbr" | "address" | "area" | "article" | "aside" | "audio" | "b" | "base" | "bdi" | "bdo" | "big" | "blockquote" | "body" | ... 157 more ... | FunctionComponent<...>'.
      Type 'typeof CallbackTitle' is not assignable to type 'ComponentClass<any, any>'.
        Types of property 'contextType' are incompatible.
          Type 'React.Context<any> | undefined' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Context<any> | undefined'.
            Type 'React.Context<any>' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Context<any>'.
    3 | import { darken } from 'polished';
    4 |
  > 5 | export const StyledCallbackTitle = styled(CallbackTitle)`
      |                                           ^^^^^^^^^^^^^
    6 |   padding: 10px;
    7 |   border-radius: 2px;
    8 |   margin-bottom: 4px;

ERROR in src/components/DropdownOrLabel/DropdownOrLabel.tsx:13:13
TS2786: 'Label' cannot be used as a JSX component.
  Its element type 'Component<{}, any, any> | ReactElement<WithOptionalTheme<{ title?: string | undefined; slot?: string | undefined; style?: CSSProperties | undefined; children?: ReactNode; ... 251 more ...; onTransitionEndCapture?: ((event: TransitionEvent<...>) => void) | undefined; } & Partial<...>, any>, string | ... 1 more ... | ...' is not a valid JSX element.
    Type 'Component<{}, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<{}, any, any>' is not assignable to type 'ElementClass'.
    11 |   const { Label = MimeLabel, Dropdown = SimpleDropdown } = props;
    12 |   if (props.options.length === 1) {
  > 13 |     return <Label>{props.options[0].value}</Label>;
       |             ^^^^^
    14 |   }
    15 |   return <Dropdown {...props} />;
    16 | }

ERROR in src/components/DropdownOrLabel/DropdownOrLabel.tsx:15:11
TS2786: 'Dropdown' cannot be used as a JSX component.
  Its element type 'ReactElement<WithOptionalTheme<{ options: DropdownOption[]; className?: string | undefined; placeholder?: string | undefined; onChange: (option: DropdownOption) => void; ... 4 more ...; fullWidth?: boolean | undefined; } & Partial<...>, any>, string | ... 1 more ... | (new (props: any) => Component<...>)> | Componen...' is not a valid JSX element.
    Type 'Component<{}, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<{}, any, any>' is not assignable to type 'ElementClass'.
    13 |     return <Label>{props.options[0].value}</Label>;
    14 |   }
  > 15 |   return <Dropdown {...props} />;
       |           ^^^^^^^^
    16 | }
    17 |

ERROR in src/components/JsonViewer/JsonViewer.tsx:26:13
TS2786: 'CopyButtonWrapper' cannot be used as a JSX component.
  Its instance type 'CopyButtonWrapper' is not a valid JSX element.
    The types returned by 'render()' are incompatible between these types.
      Type 'React.ReactNode' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").ReactNode'.
        Type '{}' is not assignable to type 'ReactNode'.
          Type '{}' is not assignable to type 'ReactPortal'.
    24 |
    25 |   render() {
  > 26 |     return <CopyButtonWrapper data={this.props.data}>{this.renderInner}</CopyButtonWrapper>;
       |             ^^^^^^^^^^^^^^^^^
    27 |   }
    28 |
    29 |   renderInner = ({ renderCopyButton }) => {

ERROR in src/components/JsonViewer/JsonViewer.tsx:117:34
TS2769: No overload matches this call.
  Overload 1 of 2, '(component: AnyStyledComponent): ThemedStyledFunction<any, ResolvedThemeInterface, any, any>', gave the following error.
    Argument of type 'typeof Json' is not assignable to parameter of type 'AnyStyledComponent'.
      Type 'typeof Json' is not assignable to type 'StyledComponent<any, any, any, never>'.
        Type 'typeof Json' is not assignable to type 'string'.
  Overload 2 of 2, '(component: "symbol" | "object" | "template" | "title" | "input" | "a" | "abbr" | "address" | "area" | "article" | "aside" | "audio" | "b" | "base" | "bdi" | "bdo" | "big" | "blockquote" | "body" | ... 157 more ... | FunctionComponent<...>): ThemedStyledFunction<...>', gave the following error.
    Argument of type 'typeof Json' is not assignable to parameter of type '"symbol" | "object" | "template" | "title" | "input" | "a" | "abbr" | "address" | "area" | "article" | "aside" | "audio" | "b" | "base" | "bdi" | "bdo" | "big" | "blockquote" | "body" | ... 157 more ... | FunctionComponent<...>'.
      Type 'typeof Json' is not assignable to type 'ComponentClass<any, any>'.
        Types of property 'contextType' are incompatible.
          Type 'React.Context<any> | undefined' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Context<any> | undefined'.
            Type 'React.Context<any>' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").Context<any>'.
    115 | }
    116 |
  > 117 | export const JsonViewer = styled(Json)`
        |                                  ^^^^
    118 |   ${jsonStyles};
    119 | `;
    120 |

ERROR in src/components/Markdown/AdvancedMarkdown.tsx:45:15
TS2786: 'part.component' cannot be used as a JSX component.
  Its element type 'ReactElement<any, any> | Component<{}, any, any> | null' is not a valid JSX element.
    Type 'Component<{}, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<{}, any, any>' is not assignable to type 'ElementClass'.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/tmp/redoc/node_modules/@types/styled-components/node_modules/@types/react/index").ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is not assignable to type 'ReactPortal'.
    43 |         );
    44 |       }
  > 45 |       return <part.component key={idx} {...{ ...part.props, ...part.propsSelector(store) }} />;
       |               ^^^^^^^^^^^^^^
    46 |     });
    47 |   }
    48 | }

ERROR in src/components/RedocStandalone.tsx:36:6
TS2786: 'ErrorBoundary' cannot be used as a JSX component.
  Its instance type 'ErrorBoundary' is not a valid JSX element.
    The types returned by 'render()' are incompatible between these types.
      Type '{} | null | undefined' is not assignable to type 'ReactNode'.
        Type '{}' is not assignable to type 'ReactNode'.
          Type '{}' is not assignable to type 'ReactPortal'.
    34 |
    35 |   return (
  > 36 |     <ErrorBoundary>
       |      ^^^^^^^^^^^^^
    37 |       <StoreBuilder spec={spec} specUrl={specUrl} options={options} onLoaded={onLoaded}>
    38 |         {({ loading, store }) =>
    39 |           !loading ? (

ERROR in src/components/SourceCode/SourceCode.tsx:20:6
TS2786: 'CopyButtonWrapper' cannot be used as a JSX component.
  Its instance type 'CopyButtonWrapper' is not a valid JSX element.
    18 |   const { source, lang } = props;
    19 |   return (
  > 20 |     <CopyButtonWrapper data={source}>
       |      ^^^^^^^^^^^^^^^^^
    21 |       {({ renderCopyButton }) => (
    22 |         <SampleControlsWrap>
    23 |           <SampleControls>{renderCopyButton()}</SampleControls>

webpack 5.73.0 compiled with 10 errors in 31657 ms
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by AlexVarchuk** [comment]
@deancn what ts version do you use in Linux? I cannot reproduce it 🙈

**Comment by AlexVarchuk** [comment]
@deancn I mean Typescript version.

**Comment by deancn** [comment]
> @deancn I mean Typescript version.

So sorry,  it's my fault.    I haven't changed the branch on Linux server.
Both Mac and Linux are built successfully. 

Let's review the code and merge, thank you again.

File: src/components/JsonViewer/JsonViewer.tsx [hunk 1]
```diff
@@ -93,26 +89,25 @@ class Json extends React.PureComponent<JsonProps> {
...
+      node?.addEventListener('click', clickListener);
+      node?.addEventListener('focus', focusListener);
+    }
+  }, [clickListener, focusListener, node]);
 
-  componentWillUnmount() {
-    this.node!.removeEventListener('click', this.clickListener);
-    this.node!.removeEventListener('focus', this.focusListener);
-  }
-}
```
Comment (Oprysk) [inline src/components/JsonViewer/JsonViewer.tsx:115]: should't we remove event listener or I missed something?

File: src/components/JsonViewer/JsonViewer.tsx [hunk 2]
```diff
@@ -93,26 +89,25 @@ class Json extends React.PureComponent<JsonProps> {
...
-  componentDidMount() {
-    this.node!.addEventListener('click', this.clickListener);
-    this.node!.addEventListener('focus', this.focusListener);
-  }
+  React.useEffect(() => {
+    if (node) {
+      node?.addEventListener('click', clickListener);
+      node?.addEventListener('focus', focusListener);
+    }
+  }, [clickListener, focusListener, node]);
```
Comment (Oprysk) [inline src/components/JsonViewer/JsonViewer.tsx:107]: this hook will be running every time component updates, is it ok?

if no, u can add useMount hook

```js
export const useMount = (callback: () => void): void => {
  useEffect(() => {
    callback();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
};
```

File: src/components/JsonViewer/JsonViewer.tsx (hunk 1 above)
Comment (AlexVarchuk) [reply src/components/JsonViewer/JsonViewer.tsx:115]: fixed. Thank for the comment.

File: src/components/JsonViewer/JsonViewer.tsx (hunk 2 above)
Comment (AlexVarchuk) [reply src/components/JsonViewer/JsonViewer.tsx:109]: I think yes. It does not work correctly with useMount hook.

**Review by Oprysk — APPROVED** [review]

File: src/components/JsonViewer/JsonViewer.tsx (hunk 2 above)
Comment (Oprysk) [reply src/components/JsonViewer/JsonViewer.tsx:109]: yeah, probably because on initial render we don't have a node

**Comment by deancn** [comment]
Can you help create a new tag for this PR.

**Pull request merged**